clients/drcachesim/simulator/tlb.cpp

```diff
@@ -132,5 +143,63 @@ tlb_t::request(const memref_t &memref_in)
     }
 }
 
+tlb_lfu_t::tlb_lfu_t(const std::string &name)
+    : tlb_t(name)
+{
+}
+
+std::string
+tlb_lfu_t::get_replace_policy() const
+{
+    return REPLACE_POLICY_LFU;
+}
+
+tlb_bit_plru_t::tlb_bit_plru_t(const std::string &name, int seed)
+    : tlb_t(name)
+    , gen_(seed == -1 ? std::random_device()() : seed)
+{
+}
+
+void
+tlb_bit_plru_t::access_update(int block_idx, int way)
+{
+    get_caching_device_block(block_idx, way).counter_ = 1;
+    for (int curr_way = 0; curr_way < associativity_; ++curr_way) {
+        if (get_caching_device_block(block_idx, curr_way).counter_ == 0) {
+            return;
+        }
+    }
+    for (int curr_way = 0; curr_way < associativity_; ++curr_way) {
+        get_caching_device_block(block_idx, curr_way).counter_ = 0;
+    }
+    get_caching_device_block(block_idx, way).counter_ = 1;
+}
+
+int
+tlb_bit_plru_t::get_next_way_to_replace(int block_idx) const
+{
+    std::vector<int> zero_counter_ways;
+    for (int way = 0; way < associativity_; ++way) {
+        caching_device_block_t &curr_block = get_caching_device_block(block_idx, way);
+        if (curr_block.tag_ == TAG_INVALID) {
+            return way;
+        }
+        if (curr_block.counter_ == 0) {
+            zero_counter_ways.push_back(way);
+        }
+    }
+    if (!zero_counter_ways.empty()) {
+        std::uniform_int_distribution<size_t> distrib(0, zero_counter_ways.size() - 1);
+        return zero_counter_ways[distrib(gen_)];
```

Comment:
Please add some unit tests exercising this new replacement scheme to clients/drcachesim/tests/cache_replacement_policy_unit_test.cpp